src/main/java/com/hillel/items_exchange/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,120 @@
+package com.hillel.items_exchange.service.impl;
+
+import com.hillel.items_exchange.dao.CategoryRepository;
+import com.hillel.items_exchange.dto.CategoryDto;
+import com.hillel.items_exchange.dto.SubcategoryDto;
+import com.hillel.items_exchange.model.Category;
+import com.hillel.items_exchange.model.Subcategory;
+import com.hillel.items_exchange.service.CategoryService;
+import com.hillel.items_exchange.service.SubcategoryService;
+import java.util.*;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+import static com.hillel.items_exchange.mapper.UtilMapper.*;
+
+@Service
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+
+    private final CategoryRepository categoryRepository;
+    private final SubcategoryService subcategoryService;
+    private final ModelMapper modelMapper;
+
+    @Override
+    public List<String> findAllCategoryNames() {
+        return categoryRepository.findAllCategoriesNames();
+    }
+
+    @Override
+    public List<CategoryDto> findAllCategoryDto() {
+        return new ArrayList<>(convertAllTo(categoryRepository.findAll(),
+                CategoryDto.class, ArrayList::new));
+    }
+
+    @Override
+    public Optional<CategoryDto> findCategoryDtoById(long id) {
+        return categoryRepository.findById(id)
+                .map(category -> convertTo(category, CategoryDto.class));
+    }
+
+    @Override
+    public CategoryDto create(CategoryDto categoryDto) {
+        final Category savedCategory = saveCategory(categoryDto);
+        return convertTo(savedCategory, CategoryDto.class);
+    }
+
+    @Override
+    public CategoryDto update(CategoryDto categoryDto) {
+        final Category updatedCategory = saveCategory(categoryDto);
+        return convertTo(updatedCategory, CategoryDto.class);
+    }
```

Comment:
`create(CategoryDto categoryDto)` and `update(CategoryDto categoryDto)` methods have the same code

Reply:
Fixed